### ISSUE TITLE
Into space 2 - multiple new issues

### ISSUE THREAD

**Issue by Kz-92** [ruffle-rs/ruffle#10033]
### Describe the bug

This game now works on latest Ruffle! now that we can get past the menus, we can see what else is needed.
I've made some videos
the first launch is always the same
flash
(it actually still loads ads, huh, cool)

https://user-images.githubusercontent.com/103858019/224587139-d9377990-db8c-4b67-8bfe-da93cf9f404e.mp4

Ruffle

https://user-images.githubusercontent.com/103858019/224587156-9877ec9c-cb67-4adf-83cc-a268cad78471.mp4

the issues are
the tutorial checks appear too early and are already checked before you actually do the action/before the tutorial appears 
the engines are always on even when not supposed to be, even in menus
text is gone on some areas
the rocket head acts as if it has a payload and is always too heavy to fly, even though it should be able to fly
ads don't play on ruffle (free adblock by CORS, I assume) 😆 
the lower rocket after falling away rests on the ground or explodes on flash, with ruffle, it falls through the floor

### Expected behavior

the game should act like it does on  the video for flash 

### Affected platform

Desktop app

### Operating system

Windows 10

### Additional information

This issue also affects into space 3, as its basically the same game
[603934_IntoSpace2.swf.zip](https://github.com/ruffle-rs/ruffle/files/10952667/603934_IntoSpace2.swf.zip)
[intospace3xmasstory_v2.zip](https://github.com/ruffle-rs/ruffle/files/10952670/intospace3xmasstory_v2.zip)

**Comment by Lord-McSweeney** [comment]
Error at menu: `TypeError: Error #1009: Cannot access a property or method of a null object reference. (accessing field: addEventListener)`
Error when changing tutorial stages: `RustError("Cannot coerce Object(ScriptObject(ScriptObject { name: com.pblabs.engine.core::InputManager, ptr: ... })) to an flash.display::InteractiveObject")`

**Comment by evilpie** [comment]
This game needs at least:
- XML.nodeKind()
- XML.namespace()
- A Namespace object with uri/prefix/toString
- XML.hasOwnProperty("@attribute")

**Comment by n0samu** [comment]
With the latest PRs, this issue is fixed:
> the engines are always on even when not supposed to be, even in menus

The other issues are still occurring, and I noticed 2 more:
- The game does not save progress between sessions.
- After you unlock the missions, clicking the missions menu goes to a blank white screen and softlocks the game.

**Comment by n0samu** [comment]
>  the tutorial checks appear too early and are already checked before you actually do the action/before the tutorial appears

This issue seems to have been fixed recently.

**Comment by Kz-92** [comment]
as noticed by nosamu, 2 or 3 speed boosters freeze the game
https://github.com/ruffle-rs/ruffle/assets/103858019/778b53ed-a537-4d02-a83f-35fe86cd0b7e

the rocket boosters unlock weird
ususally they pop up instantly notifying the unlock, but in this case, they do it on trying to buy it, and locks you from actually using it, ahah
```
ERROR ruffle_core::loader: Error during movie loading: FetchError("failed to resolve host name")
ERROR ruffle_core::avm2::events: Error dispatching event EventObject(EventObject { type: "mouseDown", class: flash.events::MouseEvent, ptr: 0x1d3aa75ab88 }) to handler FunctionObject(FunctionObject { ptr: 0x1d3aa97c358 }) : TypeError: Error #1034: Type Coercion failed: cannot convert com.pblabs.engine.core::InputManager@00000000000 to flash.display.InteractiveObject.
ERROR ruffle_core::avm2::events: Error dispatching event EventObject(EventObject { type: "click", class: flash.events::MouseEvent, ptr: 0x1d3b34ad788 }) to handler FunctionObject(FunctionObject { ptr: 0x1d3b4b8bea8 }) : RustError("Resolved parameter type void is unresolvable, not a class, null, or undefined")
ERROR ruffle_core::avm2::events: Error dispatching event EventObject(EventObject { type: "enterFrame", class: flash.events::Event, ptr: 0x1d3b5bd2038 }) to handler FunctionObject(FunctionObject { ptr: 0x1d3a6ccadc8 }) : RustError("Resolved parameter type void is unresolvable, not a class, null, or undefined")
ERROR ruffle_core::avm2::events: Error dispatching event EventObject(EventObject { type: "enterFrame", class: flash.events::Event, ptr: 0x1d3b5bd0cf8 }) to handler FunctionObject(FunctionObject { ptr: 0x1d3a6ccadc8 }) : Error: Not at root!
```
these are what I get spammed

**Comment by Kz-92** [comment]
oh wow, encoded funny, hold on

**Comment by Kz-92** [comment]
with the addition of filters, the Barbarian games logo's blur/ glow is visually messed up

https://github.com/ruffle-rs/ruffle/assets/103858019/7b44b87d-a794-4093-b725-0a6dcb6d7974

**Comment by n0samu** [comment]
It might just be a result of the glow filter being unimplemented in Ruffle, so only the blur is shown. That rectangular shape is a little suspicious though, so I'm not sure.

**Comment by n0samu** [comment]
After reading some discussion on Discord, I'm pretty sure the wrong-shaped blur is due to filter rect expansion not being implemented yet, so we should check again whenever a PR for that is up.

**Comment by n0samu** [comment]
With the latest version of #11878, the glow filter looks perfect! 
![image](https://github.com/ruffle-rs/ruffle/assets/71368227/77227a7a-1659-497a-a215-6eda4cc0c51a)

**Comment by n0samu** [comment]
Progress is saving now thanks to #12081.

The error preventing buying the booster upgrade was:
```
ERROR ruffle_core::avm2::events: Error dispatching event EventObject(EventObject { type: "click", class: flash.events::MouseEvent, ptr: 0x24c4d51ac30 }) to handler FunctionObject(FunctionObject { ptr: 0x24c4550b1c0 }) : RustError("Failed to lookup class Multiname { ns: Single(Namespace(Namespace(\"\"))), name: Some(\"void\"), param: None, flags: MultinameFlags(0x0) } during coercion")
```
This has been fixed by #11520.

Also, I tested in the latest nightly and getting 3 boosters doesn't freeze the game anymore. So I think the game is working perfectly now! 🎉
